src/NSwag.SwaggerGeneration.AspNetCore/OperationParameterProcessor.cs

```diff
@@ -0,0 +1,334 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="WebApiToSwaggerGeneratorSettings.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using NJsonSchema;
+using NJsonSchema.Generation;
+using NJsonSchema.Infrastructure;
+using NSwag.SwaggerGeneration.Processors;
+using NSwag.SwaggerGeneration.Processors.Contexts;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    internal class OperationParameterProcessor : IOperationProcessor
+    {
+        private readonly AspNetCoreToSwaggerGeneratorSettings _settings;
+
+        public OperationParameterProcessor(AspNetCoreToSwaggerGeneratorSettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>Processes the specified method information.</summary>
+        /// <param name="operationProcessorContext"></param>
+        /// <returns>true if the operation should be added to the Swagger specification.</returns>
+        public async Task<bool> ProcessAsync(OperationProcessorContext operationProcessorContext)
+        {
+            if (!(operationProcessorContext is AspNetCoreOperationProcessorContext context))
+            {
+                return false;
+            }
+
+            var httpPath = context.OperationDescription.Path;
+            var parameters = context.ApiDescription.ParameterDescriptions;
+
+            var methodParameters = context.MethodInfo.GetParameters();
+
+            foreach (var apiParameter in parameters.Where(p => p.Source != null))
+            {
+                var parameter = methodParameters.FirstOrDefault(m => m.Name == apiParameter.Name);
+                if (parameter == null || parameter.GetCustomAttributes().Any(a => a.GetType().Name == "SwaggerIgnoreAttribute"))
+                {
+                    continue;
+                }
```

Comment:
Should we use https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/ApiExplorer/ApiParameterDescription.cs#L43

Reply:
This was added more recently (as part of 2.0.0). Seems easier to just look it up rather than reflection + fallback.